### ISSUE TITLE
multi-label classification / paperswithcode dataset

### ISSUE THREAD

**Issue by x0rzkov** [Novetta/adaptnlp#86]
Hi guys,

Hope you are all well !

I was wondering if **adaptnlp** can handle multi-label classification with 1560 labels.

More precisely, I would like to apply it to paperswithcode dataset where `labels` are called `tasks`.

Refs:
- [Repository](https://github.com/paperswithcode/paperswithcode-data) 
- [Dataset](https://paperswithcode.com/media/about/papers-with-abstracts.json.gz)

Thanks for any insights or inputs on that.

Cheers,
X

**Comment by aychang95** [comment]
Hi @x0rzkov, thanks for the interesting question. Short is answer is I don't see why it wouldn't work!

I've never trained a sequence classifier on the 1000+ labels, but I have worked with 100+ labels and have seen reasonably accurate results using some of the smaller base transformers models like "bert-base-cased". The `EasySequenceClassifier` let's you train pre-trained models with a sequence classification prediction head that has a variable `nn.Linear` layer that outputs the final logits which should easily match the number of labels you provide via. your dataset.

 I would check out the sequence classification documentation [here](https://novetta.github.io/adaptnlp/tutorial/sequence-classification.html) which goes into detail on how to train on your own dataset (you can you slightly formatted CSV data files that hold your paperwithcode datasets).

Please let me know if you have any questions or feedback on the approach!

**Comment by x0rzkov** [comment]
Thanks @aychang95 for your reply :-)

Would you help me to implement that ? I can create a google colab ?! ^^

**Comment by aychang95** [comment]
@x0rzkov not a problem and sure I'd be happy to help out!

It would be an interesting use case that we could also link to in the tutorials, that is if you are okay with of course

**Comment by x0rzkov** [comment]
Yes, of course that would be awesome.

### download the dataset
```bash
wget -nc https://paperswithcode.com/media/about/papers-with-abstracts.json.gz
gunzip papers-with-abstracts.json.gz
```

### Excerpt
```json
  {
    "paper_url": "https://paperswithcode.com/paper/understanding-the-semantic-intent-of-natural",
    "arxiv_id": null,
    "title": "Understanding the Semantic Intent of Natural Language Query",
    "abstract": "",
    "url_abs": "https://www.aclweb.org/anthology/I13-1063/",
    "url_pdf": "https://www.aclweb.org/anthology/I13-1063",
    "proceeding": "IJCNLP 2013 10",
    "authors": [
      "Juan Xu",
      "Qi Zhang",
      "Xuanjing Huang"
    ],
    "tasks": [],
    "date": "2013-10-01"
  }
```

Do you have a format specification so I can pre-process their dump ?

Btw, my twitter handle is [x0rzkov](https://twitter.com/x0rzkov) or `deepocrates` on telegram for faster DM, and in the intent of not polluting this thread ^^.

**Comment by x0rzkov** [comment]
Here is the google colab link:
https://colab.research.google.com/drive/1XCkcy42VbdEfLkAJOD1gdKAX9LL58m3h?usp=sharing

**Comment by aychang95** [comment]
Great! As for the format specifications, the sequence classification data can be either a two column CSV: text column and the label column or a two column`datasets.Dataset` object from the datasets library [here](https://github.com/huggingface/datasets). 

The text column can be a string and you can call the column name whatever you'd like...I think in this case it's "content". You specify this in the `text_col_nm` param when running the `EasySequenceClassifier.train()` method.

The label column needs to be an `int` value and the column name in this case is "tasks" I believe. The labels would also be in the set [0:1559]. You specify "tasks" in the `label_col_nm` when running the train method.

Example:
content, tasks
"Paper content foobar0", 0
"Paper content foobar1", 1

Where 0 is mapped to maybe the "Transfer Learning" label" and 1 is "Text Generation".

**Comment by x0rzkov** [comment]
Hi @aychang95 

Sorry, I do not get the specifications clearly, can you update the colab that I prepared ?

Cheers,
X
